Core.Collectors/IO/AzureHelpers.cs

```diff
@@ -269,5 +275,51 @@ public static string GenerateNotificationMessage(CloudBlob blob)
             string path = blobPath.Split($"{container}/").Last();
             return Tuple.Create(container, path);
         }
+
+        public static CloudStorageAccount GetStorageAccount(string storageConnectionEnvironmentVariable)
+        {
+            string stagingBlobConnectionString = Environment.GetEnvironmentVariable(storageConnectionEnvironmentVariable);
+            return CloudStorageAccount.Parse(stagingBlobConnectionString);
+        }
+
+        public static async Task<CloudStorageAccount> GetStorageAccountUsingMsiAsync(string storageAccountNameEnvironmentVariable, ITelemetryClient telemetryClient, bool isQueue = false)
+        {
+            string storageAccountName = Environment.GetEnvironmentVariable(storageAccountNameEnvironmentVariable);
+            string resource = isQueue ? GetQueueResource(storageAccountName) : GetBlobResource(storageAccountName);
+            StorageCredentials creds = await GetStorageCredentails(resource, telemetryClient).ConfigureAwait(false);
+            CloudStorageAccount storageAccount = new CloudStorageAccount(creds, storageAccountName, EndPointSuffix, true);
+            return storageAccount;
+        }
+
+        private static async Task<StorageCredentials> GetStorageCredentails(string resource, ITelemetryClient telemetryClient)
+        {
+            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            string token = await azureServiceTokenProvider.GetAccessTokenAsync(resource).ConfigureAwait(false);
+            var jwt = new JwtSecurityToken(token);
+            tokenExpiration = jwt.ValidTo;
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                { "Resource", resource },
+                { "TokenValidTo", tokenExpiration.ToString()}
+            };
+            telemetryClient.TrackEvent("MsiTokenGeneration", properties);
```

Comment:
Now that these is a chance the telemetryCleint is null (you default to null in one of the methods). This line will crash with a NRE unless you change this to:
```
telemetryClient?.
```
Note the "?". I don't like such coding much but this will likely save you from a "compatibility issue" and you can fix this later on (e.g., enforce this once you update everywhere).

Reply:
Actually, I enforced to pass TelemetryClient so we dont run into NRE issue for StorageQueues. But yes, I went ahead and added ? check just to be safe